kong/resty/dns/client.lua

```diff
@@ -720,6 +728,7 @@ local function individualQuery(qname, r_opts, try_list)
   local result
   result, err = r:query(qname, r_opts)
   if not result then
+    close_socks(r)
```

Comment:
this only destroys in case of error, shouldn't we always destroy it?

Reply:
Sorry for missing it here